generate.sh

```diff
@@ -1,4 +1,40 @@
-java -jar ~/openapi/openapi-generator-cli.jar generate -i ../framework/openapi.yaml -g go -o pkg/galasaapi --additional-properties=packageName=galasaapi
+#!/usr/bin/env bash
+
+# Where is this script executing from ? Lets find out...
+BASEDIR=$(dirname "$0");pushd $BASEDIR 2>&1 >> /dev/null ;BASEDIR=$(pwd);popd 2>&1 >> /dev/null
+export ORIGINAL_DIR=$(pwd)
+cd "${BASEDIR}"
+
+# Allow an environment variable to over-ride the location of the openapi generator cli tool
+# otherwise set a default.
+if [[ -z $OPENAPI_GENERATOR_CLI_JAR ]]; then
+    export OPENAPI_GENERATOR_CLI_JAR=~/openapi/openapi-generator-cli.jar
+    echo "The location of the open generator jar is not specified. Defaulting it to ${OPENAPI_GENERATOR_CLI_JAR}"
+    echo "Set the OPENAPI_GENERATOR_CLI_JAR environment variable to over-ride this setting."
+    exit 1
+fi
+
+# OpenAPigen should be 6.0.1
+
+# Make sure that the openapi generator cli tool is present where we expect.
+if [[ ! -e ${OPENAPI_GENERATOR_CLI_JAR} ]]; then
+    echo "The openapi generator is not found at ${OPENAPI_GENERATOR_CLI_JAR}."
+    echo "Download it and set the OPENAPI_GENERATOR_CLI_JAR environment variable to point to it."
+    exit 1
+fi
+
+export OPENAPI_YAML_FILE="../framework/openapi.yaml"
+
+if [[ ! -e ${OPENAPI_YAML_FILE} ]]; then  
+    echo "This build requires that the galasa.dev/framework repository is checked-out to ${BASEDIR}"
+    exit 1
+fi
+
+java -jar ${OPENAPI_GENERATOR_CLI_JAR} generate \
+-i ${OPENAPI_YAML_FILE} \
+-g go \
+-o pkg/galasaapi \
+--additional-properties=packageName=galasaapi
 
 rm pkg/galasaapi/go.mod
```

Comment:
Why do we do this ?

Reply:
These appear to be clearing the NON-submodule project, as these wouldnt exsist in a submodule component. Very strange. We need to work out what exactly is required for this, as this doesnt feel correct at all

Reply:
@Jimbo4794 I seem to remember when we were trying to rebuild the cli image to use it in the regression testing cronjob we were manually going through the build steps for the cli, and as part of that kept deleting those two directories. Trying to remember why..